examples/common/lib/utils.sh

```diff
@@ -67,21 +67,49 @@ function wait_for_healthy_shard_primary() {
 	fi
 }
 
+# Wait for the shard primary tablet's VReplication engine to open.
+# There is currently no API call or client command that can be specifically used
+# to check the VReplication engine's status (no vars in /debug/vars etc. either).
+# So we use the Workflow listall client command as the method to check for that
+# as it will return an error when the engine is closed -- even when there are
+# no workflows.
+function wait_for_shard_vreplication_engine() {
+        if [[ -z ${1} || -z ${2} ]]; then
+                fail "A keyspace and shard must be specified when waiting for the shard primary tablet's VReplication engine to open"
+        fi
+        local keyspace=${1}
+        local shard=${2}
+        local wait_secs=90
+
+        for _ in $(seq 1 ${wait_secs}); do
+                if vtctlclient --server=localhost:15999 Workflow -- "${keyspace}" listall &>/dev/null; then
+                        break
+                fi
```

Comment:
`listall` here exits with `1` or another value than `0` if the list is not valid?

Reply:
I'm not sure what you mean by "the list is not valid", but:
```
$ vtctlclient Workflow -- commerce listall; echo $?
Workflow Error: rpc error: code = Unknown desc = no primary found for shard commerce/0
1

$ vtctlclient Workflow -- commerce listall; echo $?
Workflow Error: rpc error: code = Unknown desc = TabletManager.VReplicationExec on zone1-0000000100 error: vreplication engine is closed: vreplication engine is closed
1

$ vtctlclient Workflow -- foobar listall; echo $?
Workflow Error: rpc error: code = Unknown desc = node doesn't exist: /vitess/global/keyspaces/foobar/shards/
1

$ vtctlclient Workflow -- commerce listall; echo $?
No workflows found in keyspace commerce
0
```